source/_posts/2023-09-06-release-20239.markdown

```diff
@@ -114,65 +105,128 @@ It will show exactly what the entity will look like with the current settings.
-Temporary screenshot that needs to be replaced.
+<img src='/images/blog/2023-09/template-sensor.png'>
+Screenshot of creating a template sensor.
 </p>
 
 
-## Actions for template entities
+## Actions for trigger template entities
+
+Trigger template entities can now have an action block that is executed after the trigger but before the entities are rendered.
+
+This will allow users to fetch things like calendar events and weather forecasts using services with responses and use them in their template entities. You can even use OpenAI to set the value of your entities now!
+
+All variables from the action block will be available in the template.
+
+An example that checks at midnight if you have something on your calendar tomorrow and sets a binary sensor:
+
+{% raw %}
+
+```yaml
+template:
+  - trigger:
+      - platform: time
+        at: "00:00:00"
+    action:
+      - service: calendar.list_events
+        target:
+          entity_id: calendar.calendar
+        data:
+          start_date_time: "{{ today_at() + timedelta(days=1) }}"
+          duration:
+            hours: 24
+        response_variable: events
+    binary_sensor:
+      - name: Calendar events tomorrow
+        state: "{{ events.events is defined and events.events | count > 0 }}"
+```
+
+{% endraw %}
 
-Trigger template entities can now have an action block that is executed after the trigger but before the entities are rendered. All variables from the action block will be available during the entity rendering. This will allow users to fetch things like weather forecast or calendar events and use them in their template entities. Long term goal is to align the trigger template entities with automation syntax so we can support things like tracing too.
 
 ## Weather forecast service
 
-We added a weather forecast service, that returns the weather forecast.
+We added a new service for weather entities, `weather.get_forecast`, that responds with the weather forecast.
 
 Previously you would have to look at the `forecast` attribute of a weather entity to get the forecast, now this can be done by calling a new service, `weather.get_forecast`, that will respond with the forecast.
-You can also specify what type of forecast you want, daily or do you want to know it for every hour? This means you no longer need separate entities for every forecast type, but it can all be done in 1 entity.
-
-If an entity supports multiple forecasts, you can choose which one you want to see in the more info dialog.
-In the weather card in your dashboard, you can also now pick what kind of weather forecast you want to show in that card.
+You can also specify what type of forecast you want, do you want to see the forecast by day or do you want to know it for every hour? This means you no longer need separate entities for every forecast type, but it can all be done in 1 entity.
```

Comment:
```suggestion
You can also specify what type of forecast you want. Do you want to see the forecast by day or do you want to know it for every hour? This means you no longer need separate entities for every forecast type, but it can all be done in 1 entity.
```